src/l1000geom/wlsr.py

```diff
@@ -0,0 +1,224 @@
+"""WLSR placement with TPB as parent volume and TTX as daughter"""
+
+from __future__ import annotations
+
+import numpy as np
+import pyg4ometry.geant4 as g4
+
+from .profiles import (
+    make_inner_wlsr_profiles,
+    make_outer_wlsr_profiles,
+)
+
+
+def _add_wls_surfaces(materials, reg, tpb_pv, tetratex_pv, lar_pv, prefix=""):
+    """Add optical border surfaces for WLS layers."""
+    g4.BorderSurface(
+        f"bsurface_{prefix}tpb_ttx",
+        tpb_pv,
+        tetratex_pv,
+        materials.surfaces.wlsr_tpb_to_tetratex,
+        reg,  # Fixed: changed from to_tetratex
+    )
+    g4.BorderSurface(f"bsurface_{prefix}lar_tpb", lar_pv, tpb_pv, materials.surfaces.lar_to_tpb, reg)
+    g4.BorderSurface(f"bsurface_{prefix}tpb_lar", tpb_pv, lar_pv, materials.surfaces.lar_to_tpb, reg)
+
+
+def place_inner_wlsr_in_argon(
+    materials,
+    registry,
+    lar_cavity_lv,
+    lar_cavity_pv,
+    neckradius,
+    tubeheight,
+    totalheight,
+    curvefraction,
+    wls_height,
+    inner_z,
+    inner_r,
+    outer_z,
+    outer_r,
+):
+    """
+    Place inner WLS layers in the underground argon
+    """
+    result = make_inner_wlsr_profiles(
+        neckradius, tubeheight, totalheight, curvefraction, wls_height, inner_z, inner_r
+    )
+
+    (
+        tpb_outer_z,
```

Comment:
I do not particularly like this result from the code style, but there is no way to make it nicer, I guess. Maybe refactoring into returning a small object instead of tuples might help? But not so important

Reply:
maybe it can return a `namedtuple`?